src/firebolt/utils/util.py

```diff
@@ -169,6 +172,28 @@ def _print_error_body(resp: Response) -> None:
         pass
 
 
+def raise_errors_from_body(resp: Response) -> None:
+    """Process error in response body.
+
+    Args:
+        resp (Response): HTTP response
+    """
+    to_raise = None
+    try:
+        decoded = resp.json()
+        if "errors" in decoded and len(decoded["errors"]) > 0:
+            # Raise later to avoid catching it in the except block
+            to_raise = FireboltStructuredError(decoded)
+
+    except Exception:
```

Comment:
Let's be more specific with errors here. I would suggest to only try catch json parsing part with `JSONDecodeError`, everything else shouldn't raise an error

Reply:
this way we can also raise directly from the try catch body, yet that's up to you

Reply:
Yeah, I thought of this at first. But then IMO we really shouldn't be raising any error here apart from the one we've successfully parsed. If the parsing breaks or some other error causes an issue and we don't catch it we'll obscure the actual error here.
I think somewhat awkward code here is a small price to pay for being sure this doesn't happen.

Reply:
Ok, that's probably better. Please add this explanation to the function docstring